src/main/java/de/tum/in/www1/artemis/service/InternalUrlService.java

```diff
@@ -0,0 +1,108 @@
+package de.tum.in.www1.artemis.service;
+
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import de.tum.in.www1.artemis.domain.VcsRepositoryUrl;
+
+@Profile("bamboo")
+@Service
+public abstract class InternalUrlService {
```

Comment:
Because this is only useful for Bamboo, can we include the word "Bamboo" in the name of this class?

Reply:
Oh this is not correct, thanks! The internalUrlService is used by Bamboo and Jenkins with their own Implementation.